engines/account/app/services/account/signup_service.rb

```diff
@@ -1,19 +1,19 @@
 module Account
   class SignupService
 
-    # @param email [String]
-    def initialize(email)
-      @email = email
-      @signup = Signup.find_by_email(email) || Signup.new(email: email)
+    def initialize
     end
 
+    # @param email [String]
     # @return [Account::Signup]
-    def execute
-      if @signup.save
-        SignupMailer.confirm_email(@signup).deliver
+    def create!(email)
+      signup = Signup.new(email: email)
```

Comment:
`Signup.find_by_email(email) || Signup.new(email: email)`

to keep the same token for the same email address